tests/agent/test_event_handler.py

```diff
@@ -20,76 +46,152 @@
-    )
-    event = RunMissingnessDataAssistantEvent(
-        datasource_name="test-datasource",
-        data_asset_name="test-data-asset",
-    )
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+    def test_event_handler_raises_on_no_version_implementation(
+        self, mock_context, mocker: MockerFixture
+    ):
+        gx_major_version = mocker.patch(
+            "great_expectations_cloud.agent.event_handler._GX_MAJOR_VERSION"
+        )
+        gx_major_version.return_value = "NOT_A_REAL_VERSION"
 
-    handler.handle_event(event=event, id=correlation_id)
+        handler = EventHandler(context=mock_context)
 
-    action.assert_called_with(context=mock_context)
-    action().run.assert_called_with(event=event, id=correlation_id)
+        with pytest.raises(NoVersionImplementationError):
+            handler.get_event_action(DummyEvent)  # type: ignore[arg-type]  # Dummy event only used in testing
 
 
-def test_event_handler_handles_run_onboarding_data_assistant_event(
-    mock_context, mocker: MockerFixture
-):
-    action = mocker.patch(
-        "great_expectations_cloud.agent.event_handler.RunOnboardingDataAssistantAction"
-    )
-    event = RunOnboardingDataAssistantEvent(
-        datasource_name="test-datasource", data_asset_name="test-data-asset"
-    )
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+class DummyEvent(EventBase):
+    type: Literal["event_name.received"] = "event_name.received"
 
-    handler.handle_event(event=event, id=correlation_id)
 
-    action.assert_called_with(context=mock_context)
-    action().run.assert_called_with(event=event, id=correlation_id)
+class DummyAction(AgentAction[Any]):
+    @override
+    def run(self, event: Event, id: str) -> ActionResult:
+        return ActionResult(id=id, type="DummyAction", created_resources=[])
 
 
-def test_event_handler_handles_run_checkpoint_event(mock_context, mocker: MockerFixture):
-    action = mocker.patch("great_expectations_cloud.agent.event_handler.RunCheckpointAction")
-    event = RunCheckpointEvent(
-        checkpoint_id="3ecd140b-1dd5-41f4-bdb1-c8009d4f1940",
-        datasource_names_to_asset_names={"Data Source name": {"Asset name"}},
+class TestEventHandlerRegistry:
+    @pytest.mark.parametrize(
+        "version",
+        [
+            "0",
+            "1234567890",  # Version that doesn't exist already in the map
+        ],
     )
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+    def test_register_event_action(self, version: str):
+        register_event_action(version, DummyEvent, DummyAction)
+        assert _EVENT_ACTION_MAP[version][DummyEvent.__name__] == DummyAction
 
-    handler.handle_event(event=event, id=correlation_id)
+        del _EVENT_ACTION_MAP[version][DummyEvent.__name__]  # Cleanup
```

Comment:
This is a dangerous way to undo this patch because it won't be hit if the assert above it is triggered.
I would suggest using `pytest` `monkeypatch.setitem()`

https://docs.pytest.org/en/latest/how-to/monkeypatch.html#monkeypatching-dictionaries

Reply:
What about `mocker.patch.dict` instead? Just to keep it consistent by using `mocker` https://pytest-mock.readthedocs.io/en/latest/usage.html

Reply:
Fixed - good call on the suggestion.